.github/workflows/test.yml

```diff
@@ -0,0 +1,25 @@
+name: tests
+
+on: [push]
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Set up Go
+        uses: actions/setup-go@v3
+        with:
+          go-version: "1.20"
+
+      - name: Set up GCC
```

Comment:
Do we want to test against the ubuntu compiler that's bundled with ubuntu-latest instead, it might be more representative of common customer configs?

Reply:
I'm not really worried. If someone finds a bug we can add another compiler.

Reply:
What I mean is, we could leave this out until we need it.

Reply:
I believe github actions ubuntu-latest doesn't ship with a C toolchain, you need to apt-ge install it. I figured this would be faster.

Reply:
Oh, the images are surprisingly substantial: https://github.com/actions/runner-images/blob/main/images/linux/Ubuntu2204-Readme.md

Reply:
huh, I'm reasonably sure I used to have to install gcc on github. but it seems to work! less stuff!